plugins/unicodelib/unicodelib.cpp

```diff
@@ -719,6 +724,76 @@ unsigned unicodeEditDistanceV4(UnicodeString & left, UnicodeString & right, unsi
     return da[mask(leftLen-1)][rightLen-1];
 }
 
+void normalizationFormCheck(UnicodeString & source, const char * form)
+{
+    UErrorCode errorCode = U_ZERO_ERROR;
+    if (form[2] == 'C')
+    {
+        const Normalizer2 * no = Normalizer2::getInstance(NULL, "nfc", UNormalization2Mode::UNORM2_DECOMPOSE, errorCode);
+        source = no->normalize(source, errorCode);
+    }
+    else if (form[2] == 'D')
+    {
+        const Normalizer2 * no = Normalizer2::getInstance(NULL, "nfc", UNormalization2Mode::UNORM2_DECOMPOSE, errorCode);
+        source = no->normalize(source, errorCode);
+    }
+    else if (form[3] == 'C')
+    {
+        const Normalizer2 * no = Normalizer2::getInstance(NULL, "nfkc", UNormalization2Mode::UNORM2_COMPOSE, errorCode);
+        source = no->normalize(source, errorCode);
+    }
+    else if (form[3] == 'D')
+    {
+        const Normalizer2 * no = Normalizer2::getInstance(NULL, "nfkc", UNormalization2Mode::UNORM2_DECOMPOSE, errorCode);
+        source = no->normalize(source, errorCode);
+    }
+    if (U_FAILURE(errorCode))
+    {
+        return;
+    }
+}
+
+unsigned findCount(UnicodeString & source, UnicodeString & sought)
+{
+    int32_t matches = 0;
+    if (source.isEmpty() || sought.isEmpty())
+    {
+        return matches;
+    }
+    int32_t idx = 0;
+    int32_t sourceLength = source.countChar32();
+    int32_t soughtLength = sought.countChar32();
+    int32_t max = source.length() - (soughtLength - 1);
+    UnicodeString temp;
+    StringCharacterIterator it(source);
+    if (sourceLength < soughtLength)
+    {
+        return matches;
+    }
+    while (idx < max)
+    {
+        if (source.char32At(idx) == sought.char32At(0))
+        {
+            source.extractBetween(idx, it.move32(soughtLength, CharacterIterator::kCurrent), temp);
+            if (!sought.compareCodePointOrder(temp))
+            {
+                matches++;
+                it.setIndex32(idx);
+                idx = it.move32(soughtLength, CharacterIterator::kCurrent);
+            }
+            else
+            {
+                idx = it.move32(1, CharacterIterator::kCurrent);
```

Comment:
This hasn't adjusted the current position back before advancing.  I suspect a search for "ABC" in the string "AABC" would fail as it currently stands.